### PR TITLE
Optimize MasterServer, add MasterRPCService

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

* Add MasterRPCServer, clean the code in MasterServer.
* Avoid NPR in MasterSchedulerService

### CODE REVIEW

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskExecuteRunnable.java [hunk 1]
```diff
@@ -116,12 +116,12 @@ private void persist(TaskEvent taskEvent) {
         int taskInstanceId = taskEvent.getTaskInstanceId();
         int processInstanceId = taskEvent.getProcessInstanceId();
 
-        TaskInstance taskInstance;
-        WorkflowExecuteThread workflowExecuteThread = this.processInstanceExecCacheManager.getByProcessInstanceId(processInstanceId);
+        Optional<TaskInstance> taskInstance;
+        WorkflowExecuteRunnable workflowExecuteThread = this.processInstanceExecCacheManager.getByProcessInstanceId(processInstanceId);
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskExecuteRunnable.java]:
```suggestion
        WorkflowExecuteRunnable workflowExecuteRunable = this.processInstanceExecCacheManager.getByProcessInstanceId(processInstanceId);
```

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskExecuteThreadPool.java [hunk 2]
```diff
@@ -83,26 +83,26 @@ public void submitTaskEvent(TaskEvent taskEvent) {
...
-            TaskExecuteThread taskExecuteThread = new TaskExecuteThread(
+            TaskExecuteRunnable taskExecuteThread = new TaskExecuteRunnable(
                     taskEvent.getProcessInstanceId(),
                     processService, workflowExecuteThreadPool,
                     processInstanceExecCacheManager,
                     dataQualityResultOperator);
             taskExecuteThreadMap.put(taskEvent.getProcessInstanceId(), taskExecuteThread);
         }
-        TaskExecuteThread taskExecuteThread = taskExecuteThreadMap.get(taskEvent.getProcessInstanceId());
+        TaskExecuteRunnable taskExecuteThread = taskExecuteThreadMap.get(taskEvent.getProcessInstanceId());
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskExecuteThreadPool.java]:
```suggestion
        TaskExecuteRunnable taskExecuteRunnable= taskExecuteThreadMap.get(taskEvent.getProcessInstanceId());
```

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/MasterServer.java [hunk 3]
```diff
@@ -191,10 +136,15 @@ public void close(String cause) {
             // close
             this.schedulerApi.close();
             this.masterSchedulerService.close();
-            this.nettyRemotingServer.close();
+            try (MasterRPCServer closeServer = this.masterRPCServer) {
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/MasterServer.java]: looks a little weird

**Review by caishunfeng — COMMENTED** [review]
Good optimization, and some review comments here.

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/MasterServer.java (hunk 3 above)
Comment (ruanwenjun) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/MasterServer.java]: Ok, I remove the try, and use `this.masterRPCServer.close()`

**Review by caishunfeng — APPROVED** [review]
LGTM

**Pull request merged**